src/index/mod.rs

```diff
@@ -79,6 +80,9 @@ pub trait Index {
 
     /// Clear the entire index.
     fn reset(&mut self) -> Result<()>;
+
+    /// Remove data vectors represented by IDs.
+    fn remove_ids(&mut self, sel: &IdSelector) -> Result<(i64)>;
```

Comment:
I am not entirely convinced that `i64` is preferred here. If the operation succeeds, the number of elements is sure to be non-negative. I'll be looking at what Faiss can output here. In the mean time, the parentheses are unnecessary: `Result<i64>`